src/librustc_middle/mir/interpret/mod.rs

```diff
@@ -454,8 +458,9 @@ impl<'tcx> AllocMap<'tcx> {
         });
         if is_generic {
             // Get a fresh ID.
-            let id = self.reserve();
-            self.alloc_map.insert(id, GlobalAlloc::Function(instance));
+            let mut alloc_map = self.alloc_map.lock();
+            let id = alloc_map.reserve();
+            alloc_map.alloc_map.insert(id, GlobalAlloc::Function(instance));
```

Comment:
`alloc_map.alloc_map` looks odd.

Reply:
it does, but it's only inside the `AllocMap` impls

Reply:
Where does it come from though?

Reply:
There is `TyCtxt::alloc_map` of `AllocMap` type, and then there's `AllocMap::alloc_map` of `FxHashMap` type.

Reply:
Hm... there might be better names but it's probably not worth renaming.^^